instrumentation/spring/spring-webflux/spring-webflux-5.3/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/v5_3/internal/HeaderUtil.java

```diff
@@ -32,6 +39,33 @@ public final class HeaderUtil {
             () ->
                 findGetHeadersMethod(
                     MethodType.methodType(List.class, Object.class))); // before spring web 7.0
+
+    // Spring Web 7+
+    MethodHandle forEach = null;
+    try {
+      forEach =
+          MethodHandles.lookup()
+              .findVirtual(
+                  HttpHeaders.class,
+                  "forEach",
```

Comment:
perhaps we should be using https://github.com/spring-projects/spring-framework/blob/92e9543ad44bb33cf6d89525e93d6afb0714f137/spring-web/src/main/java/org/springframework/http/HttpHeaders.java#L2054 instead to avoid copying the set.